### DISCUSSION TITLE
Signal Connection Dialog should allow for removing a signal parameter and replacing it with a predefined one.

### DISCUSSION THREAD

**Discussion by MichaelMacha** [godotengine/godot-proposals#11100] [Editor]
I currently have a game in which my player, when entering an Area3D, should trigger an animation in the environment (specifically a bunch of stairs ascending from the ground). This is a `body_entered` signal, which has a single parameter, `body`.

![image](https://github.com/user-attachments/assets/2b693bb4-7816-4807-8a2a-99c6fa9fb0e8)

I have all of the behavior for the stairs, rising from the ground, in an `AnimationPlayer` node. In theory, all I need to call is `play(animation_name : StringName)` when the signal is emitted. So, I'm very close to avoiding any GDScript at all.

_In my thirty years of coding, I've learned that the less code you write, and the closer you keep your definition to its use, the smaller the error area is and the more likely it is to require less maintenance, and perform its job properly as a project gets bigger. I imagine that that's equally true for games. So, it's extremely important to me in this relatively-early phase to write as few scripts as possible._

My thought is, if, in that dialog box, I can drop the `body` parameter and simultaneously add an extra parameter of `StringName` type, I can do this without a single line, using zero lines of script. That would be optimal for so many things.

However, when I increase the argument unbinding count to one, I lose the ability to add an additional parameter; and the best I can do is write a script for it myself. This is rather unfortunate, as I feel like this should be trivial... just unbind one argument, and bind a `StringName`, then send it.

As far as I can tell, this is currently impossible without writing code. Is there a reason for this which I'm not seeing? I've reviewed the C++, but it's unfortunately not commented and a little difficult for me to fully understand.

**Comment by AThousandShips** [comment]
Thank you for your proposal! There's already an open proposal for this however:
* https://github.com/godotengine/godot-proposals/issues/7542

And we prefer to keep things in one place, so closing this

See there for more details and please add any further details you feel might be missing

**Reply by MichaelMacha** [reply]
Crap, I just realized I've even posted over there already. I have no memory of that... big project, I guess!

Apologies for opening this in two places.

**Reply by AThousandShips** [reply]
No worries it's easy to lose track of things
